hepdata/modules/records/utils/records_update_utils.py

```diff
@@ -0,0 +1,96 @@
+import datetime
+import requests
+import math
+
+from hepdata.modules.records.utils.doi_minter import generate_dois_for_submission
+from hepdata.modules.submission.api import get_latest_hepsubmission
+from hepdata.modules.records.utils.workflow import update_record
+from hepdata.modules.inspire_api.views import get_inspire_record_information
+from hepdata.ext.elasticsearch.api import index_record_ids
+from hepdata.modules.email.api import notify_publication_update
+
+
+def update_record_info(inspire_id, recid='', send_email=False):
+    inspire_id = inspire_id.replace("ins", "")
+
+    if recid == '':
+        hep_submission = get_latest_hepsubmission(inspire_id=inspire_id)
+        if hep_submission is None:
+            print("Failed to retrieve hep submission for {0}".format(inspire_id))
+            return
+        recid = hep_submission.publication_recid
+
+    updated_record_information, status = get_inspire_record_information(inspire_id)
+
+    if status == 'success':
+        record_information = update_record(recid, updated_record_information)
```

Comment:
It could happen that some more obscure fields of an INSPIRE record are updated, but we're not using them, so there is no need to update the HEPData record.  Since we don't want to reindex or push information to DataCite unnecessarily, we should check for this case and only update the HEPData record if a relevant field has changed, e.g. replace the line above with something like:
```python
record_information = get_record_by_id(recid)
same_information = True
for key, value in updated_record_information.items():
    if key not in record_information or record_information[key] != value:
        same_information = False
        break
if same_information:
    return
record_information = update_record(recid, updated_record_information)
```

Reply:
I used exactly this code, I hope that's ok.